### DISCUSSION TITLE
Tasmota ESP32-V14.4 to Shelly downgrade

### DISCUSSION THREAD

**Discussion by chrkir** [arendst/Tasmota#22630] [Support request] [answered]
I didn't found a procedure to downgrade as described for ESP8266 with the minimal version.

**Comment by Jason2866** [comment]
Your question is very unclear. What do you want to achieve? There is no minimal firmware for esp32x based Tasmota32x firmwares.
Downgrade was and is not supported.

**Comment by chrkir** [comment]
Sorry I hit the return to fast,.

I use a Shelly 2PM as shutter control, but the calibration (open + closed) is very unstable.
For this reason I want to go back to the original Shell 2PM software, since a shelly 2.5 I have also
is very stable and the self-calibration works as expected.

I found some procedures published to restore shelly FW, but all referencing an ESP8266 with the obsolete
Tasmota-minimal.bin not available for ESP32.

So please could you give hint how to proceed ?

**Reply by sfromis** [reply]
Assuming the nonexisting "Shelly 2PM" was meant as "Shelly Plus 2PM".....

Haven't looked into this, any caveats, or how to use, but this link does look like original firmware:
https://updates.shelly.cloud/update/Plus2PM
Not sure if it even could be suitable for "cold turkey" flashing.

Anyway, `tasmota-minimal.bin.gz` for ESP8266 is completely unrelated to ideas about going back to Shelly firmware, that would instead be:
https://github.com/yaourdt/tasmota-to-mgos
AFAIK, no similar set exists for ESP32-based devices.

**Reply by Jason2866** [reply]
No, the update packages can not be used for initial flashing.

**Comment by Jason2866** [comment, marked as answer]
There is no way back to Shelly firmware for esp32x driven devices if you do not backuped yourself.

**Reply by barbudor** [reply]
May be ask to @yaourdt if he can consider making a tasmota32-to-mgos
Or help tasmota team to solve the problems you have (sorry if you have already done that to no result)

**Reply by Jason2866** [reply]
Highly doubt it is possible to create a script back to Shelly. The Shelly partition scheme is gone and the additional partitions with Shelly specific device keys, data and certs.
As clearly written in the Tasmota Shelly Convert repo "There is no way back"
